common/src/main/java/io/honeycomb/opentelemetry/EnvironmentConfiguration.java

```diff
@@ -230,36 +234,23 @@ public static void enableOTLPTraces() {
 
         // if we have an API Key, add it to the header
         if (isPresent(apiKey)) {
-            String header = String.format("%s=%s", HONEYCOMB_TEAM_HEADER, apiKey);
-            if (isLegacyKey(apiKey)) {
-                // if the key is legacy, add dataset to the header
-                if (isPresent(dataset)) {
-                    header += String.format(",%s=%s", HONEYCOMB_DATASET_HEADER, dataset);
-                } else {
-                    // if legacy key and missing dataset, warn on missing dataset
-                    System.out.printf("WARN: %s%n", getErrorMessage("dataset", HONEYCOMB_DATASET));
-                }
-            }
-            System.setProperty("otel.exporter.otlp.traces.headers", header);
+            System.setProperty("otel.exporter.otlp.traces.headers", headersToString(headers));
         } else {
             // warn on missing API Key
             System.out.printf("WARN: %s%n", getErrorMessage("API key", HONEYCOMB_API_KEY));
         }
-
     }
 
     public static void enableOTLPMetrics() {
         final String endpoint = getHoneycombMetricsApiEndpoint();
         final String apiKey = getHoneycombMetricsApiKey();
         final String dataset = getHoneycombMetricsDataset();
+        final Map<String, String> headers = getHeaders(apiKey, dataset);
```

Comment:
Oops, that's incorrect. Metrics always get the dataset header, and no warnings are show.